pkg/parquetquery/iters.go

```diff
@@ -1328,12 +1334,15 @@ func NewJoinIterator(definitionLevel int, iters []Iterator, pred GroupPredicate,
 		lowestIters:     make([]int, len(iters)),
 		peeks:           make([]*IteratorResult, len(iters)),
 		pred:            pred,
-		pool:            DefaultPool.Get,
+		pool:            DefaultPool,
 	}
 
 	for _, opt := range opts {
 		opt.applyToJoinIterator(j)
 	}
+
+	j.at = j.pool.Get()
```

Comment:
is pooling worth it if we only do it at instantiation of the iterator? maybe just drop pooling altogether for simplicity?

Reply:
It's still worth it, mentioned this in the PR description.

> Although I really wanted to eliminate pooling altogether, it's still needed to keep memory down. Pooling lets results be used across entire jobs, which clearly shows in benchmarks and in practice will help a busy querier. And we still need to keep pool items organized by size (i.e. spansets (large) vs attrs (small)).

Reply:
apologies, missed that